### PR TITLE
 test(solid-query) Add test for transition

### PR DESCRIPTION
**Pull request by jeski-bright**
Add test for running a query during transition.

~~Change `createResource` usage to only indicate the suspended loading state. Always serve the underlying data from the query state. Do not trigger mutations and refetches as often.~~

~~This is a workaround for the following issue: https://github.com/solidjs/solid/issues/1249.~~

### CODE REVIEW

**Comment by TkDodo** [comment]
Hi, let's wait for @ardeora to review your change, but in the meantime:

- please create a test case that fails with the current implementation on `main`, but doesn't fail with your fix.

thank you 🙏

**Comment by ardeora** [comment]
Thanks! Yeah this a very interesting/smart approach. I had looked at something similar before but the current implementation worked fine. Though, I'm curious as to which exact part of the current implementation is breaking transitions. I'm at work right now. But will take a deeper look at it in a few hours.

File: packages/solid-query/src/createBaseQuery.ts
Comment (ardeora) [inline packages/solid-query/src/createBaseQuery.ts:79]:
```suggestion
    if (state.fetchStatus === 'idle') {
      res()
      res = noop
    } else {
      refetch()
    }
```

We don't need a batch here

File: packages/solid-query/src/createBaseQuery.ts
Comment (ardeora) [inline packages/solid-query/src/createBaseQuery.ts:44]: Can we give this variable a more descriptive name? Maybe `resolveQueryData` would be nice? 
```suggestion
  let resolveQueryData = noop
```

File: packages/solid-query/src/createBaseQuery.ts
Comment (ardeora) [inline packages/solid-query/src/createBaseQuery.ts:50]:
```suggestion
        resolveQueryData = resolve
```

File: packages/solid-query/src/createBaseQuery.ts
Comment (ardeora) [inline packages/solid-query/src/createBaseQuery.ts:84]:
```suggestion
    resolveQueryData()
```

File: packages/solid-query/src/__tests__/transition.test.tsx
Comment (ardeora) [inline packages/solid-query/src/__tests__/transition.test.tsx:20]: One thing that still breaks (even with the current implementation) is that if you use the `Show` component instead so 

```tsx
<Show when={state.data}>
    <div>Message</div>
</Show>
```

still doesn't work 😭  There's maybe something goofy over here with solid js not resolving the reactivity correctly. I'm trying to create a minimal reproduction of where things are going wrong, but it'll take me more time

**Review by ardeora — CHANGES_REQUESTED** [review]

File: packages/solid-query/src/__tests__/transition.test.tsx
Comment (jeski-bright) [reply packages/solid-query/src/__tests__/transition.test.tsx:20]: That's true :( But I believe this must be fixed in `solid-js` core.

This however works:
```js
<div>
  <Show when={state.data}>
    <div>Message</div>
  </Show>
</div>
```

It seems that the conditional component always needs to be wrapped in another one and `Fragment` doesn't help here either.
For this one I'd expect something wrong in the transformations made by `X-plugin-solid` packages.

Comment (jeski-bright) [reply packages/solid-query/src/__tests__/transition.test.tsx:20]: This (along with the `Show`) issue seem to be fixed in `solid-js` already: https://github.com/solidjs/solid/commit/ea2c976add11b20798267c1fa218672189955ef6, though not released yet. Currently my fix breaks the `<Show when={...}` case 😂 , so how should we proceed with this?

I believe that the test case could still be useful.

Comment (ardeora) [reply packages/solid-query/src/__tests__/transition.test.tsx:20]: Ah yes! Thanks for creating an issue with Ryan about this. I built the latest version and it seems like I can get it working fine (current implementation) with the new fixes Ryan implemented. I will let solid-js push a new release out to solve this upstream issue. However, the transitions tests would be really useful. I would suggest only adding tests in this pull request for now :D Let me know if you have any questions or concerns!

Comment (ardeora) [reply packages/solid-query/src/__tests__/transition.test.tsx:20]: Our CI wont let us publish until all tests pass. So for now can we make it `it.skip`. I'll go back ad remove the skip when we have a new solid-js version out

Comment (jeski-bright) [reply packages/solid-query/src/__tests__/transition.test.tsx]: Left only the test & updated the commit message.

**Comment by ardeora** [comment]
@TkDodo  We are looking good with this PR. @jeski-bright added a new test for solid-js transitions. The upstream issue should be fixed in a few days and we can unskip the test then. Feel free to merge 😄  

Or do you suggest we hold on to this PR until the upstream issue is fixed?

**Comment by TkDodo** [comment]
@ardeora since this PR only adds a skipped test, I would wait until the upstream issue is fixed and then merge the test when we actually run it.

**Comment by jeski-bright** [comment]
Since Ryan released those changes, updated solid to 1.5.7 and enabled the test. Please double check this, as during the first run of the test suite I got 1 failing test: `useQuery › should not fetch when switching to a disabled query`, subsequent re-runs were all green...

**Comment by ardeora** [comment]
@TkDodo The changes are looking good here and `test:ci` works fine on my local machine. But CI on github actions fails. Any idea what's going on here? 😄 I can see it complaining about not finding `pnpm` on disk

**Comment by TkDodo** [comment]
Yes I fixed this in my other PR. Sorry, I should port the fix directly to main. Will do later today.

**Comment by ardeora** [comment]
No worries! Thanks for the quick reply. Thanks @jeski-bright! We'll get this merged when CI is fixed

**Comment by TkDodo** [comment]
it's merged. please update with main to make the pipeline work :)

**Review by TkDodo — COMMENTED** [review]
what is the empty file named `change` please? can we remove that?

**Comment by jeski-bright** [comment]
> what is the empty file named `change` please? can we remove that?

Sorry for that, must've been checking something. Removed.

**Review by TkDodo — APPROVED** [review]

**Pull request merged**
